codeQuality/index.md

```diff
@@ -0,0 +1,173 @@
+
+## React best practices
+
+We generally follow guidance from the [new react docs](https://react.dev/) when assessing react-specific best practices.
+
+Answers to the following questions and more can be found therein:
+
+- When to use useMemo/useCallback (hint: not unless you have to)
+- When (and when not!) to use useEffect
+- How to correctly implement useContext
+
+If a question can't be resolved by the react docs (or other authoritative source), then we default to sound programming practices (DRY/SOLID etc.). Finally, if neither of the previous stages produce a definitive answer, the preference of the person committing takes precedence.
+
+With that said, here are some other things we like to see.
+
+### Using useContext
+
+We generally prefer to use `useContext` than prop drill to pass state down multiple levels of the component hierarchy.
+
+Some exceptions to this might include:
+-> State that's needed in multiple parts of the app. If it doesn't belong in the `AppContext` provider, consider using redux.
+-> State that should persist across sessions: use local storage or the db
+-> When introducing an additional context provider is awkward/impractical or clutters the existing set of providers. In cases such as this, prop drilling is acceptable.
+
+### Passing props
+
+We prefer to pass props explicitly:
+
+```jsx
+const name = "Tim"
+const age = 100
+<MyComponent name={name} age={age} />
+```
+
+As opposed to
+
+```jsx
+<MyComponent {...{ name, age }} />
+```
+
+### Boolean logic
+
+Sometimes we render components conditionally:
+
+```jsx
+const itsMyBirthday = true;
+
+{
+  itsMyBirthday && <Card />;
+}
+```
+
+What we don't like to see is the following chaining of booleans:
+
+```jsx
+const todayIsSunday = true;
+const itsMyBirthday = true;
+const monthIsMarch = true;
+
+{
+  todayIsSunday && itsMyBirthday && monthIsMarch && <Card />;
+}
+```
+
+Wrap up your booleans into a single boolean:
+
+```jsx
+const displayCard = todayIsSunday && itsMyBirthday && monthIsMarch;
+
+{
+  displayCard && <Card />;
+}
+```
+
+Much nicer.
+
+### Object destructuring
+
+When destructuring an object that may be undefined, employ the following syntax:
+
+```jsx
+const { nativeToken } = colony || {};
+```
+
+## Javascript best practices
+
+### Explicitly scoping `if` statements
+
+Prefer brackets when writing `if` statements
+
+```js
+// Yes
+if (true) {
+  doSomething();
+}
+
+// No
+if (true) doSomething();
+```
+
+### Type casting
+
+Prefer casting via the constructor when casting numbers or strings:
+
+```js
+const a = "1";
+const b = 0;
+
+// Yes
+const c = Number(a);
+const d = String(b);
+// No
+const c = +a;
+const d = Number(0).toString();
+const d = `${b}`;
+```
+
+But via the `!!` syntax when casting booleans:
+
+```js
+const c = 0;
+// Yes
+const d = !!0;
+// No
+const d = Boolean(0);
+```
+
+## Forms best practices
+
+We use forms all the time to take user input and do something with it (usually pass it on to a saga).
```

Comment:
This should state that we prefer `react-hook-forms` over the old `Formik` going forward